pymongo/pool.py

```diff
@@ -1557,7 +1570,7 @@ def remove_stale_sockets(self, reference_generation: int) -> None:
                     if self.gen.get_overall() != reference_generation:
                         conn.close_conn(ConnectionClosedReason.STALE)
                         return
-                    self.conns.appendleft(conn)
```

Comment:
Here in `remove_stale_sockets` is the 2nd usage of Pool.connect(). Inside the latter we add the new Connection to active_conns. Is that the intended behavior in this scenario. I am unfamiliar with `self.gen` but I notice that the connection also gets added to self.available_conns.

Reply:
I've added removing the conn's cancellation context from the set of active contexts here, as suggested.